consumer/kafka/src/main/java/com/spotify/heroic/consumer/kafka/ConsumerThread.java

```diff
@@ -195,9 +245,40 @@ private void consumeOneWithRetry(final byte[] body) throws InterruptedException
 
     private boolean consumeOne(final byte[] body) {
         try {
-            schema.consume(body);
+            /* We have read something. This is a good time to check if we should prepare to commit.
+             * Why is it a good time? Because if we pause now, then send off one more consumption
+             * request, then we know that there will be _at least one_ write being finished sometime
+             * soon so we can use onFinished() to do the commit. */
+            maybePrepareToCommitConsumerOffsets();
+
+            final FutureReporter.Context consumptionContext = reporter.reportConsumption();
+
+            // Actually consume
+            final AsyncFuture<Void> future = schema.consume(body);
+
+            // FIXME: How does this work with batch writes?
+            threadLocalSequenceNumber++;
+            final Long sequenceNumber = new Long(threadLocalSequenceNumber);
```

Comment:
this is not needed, see:
https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html